src/lib.rs

```diff
@@ -157,6 +160,15 @@ use snafu::*;
 
 use auth::Auth;
 
+#[cfg(feature = "rate-limiting")]
```

Comment:
I would recommend importing `cfg_if` and using that to import instead of having to say `#[cfg(feature = "rate-limiting")]` each time.